aten/src/ATen/native/quantized/cudnn/Linear.cpp

```diff
@@ -0,0 +1,345 @@
+
+  auto run = [&](cudnn_frontend::ManagedOpaqueDescriptor plan_desc) {
+    auto workspace_size = 0;
+    auto workspace = at::empty({workspace_size}, input.options().dtype(at::kByte));
+    std::vector<void *> data_ptrs;
+    std::vector<int64_t> uids;
+    data_ptrs.reserve(10);
+    uids.reserve(10);
+    data_ptrs = {input_fp.data_ptr(), linear_output.data_ptr(),
+                                           weight_fp.data_ptr(),
+                                           requantize_multiplier_tensor.data_ptr(),
+                                           reinterpret_cast<int8_t*>(quantized_output.data_ptr())};
+    uids = {'x', 'y', 'w', 's', 'r'};
+    if (bias_.has_value()) {
+      data_ptrs.insert(data_ptrs.end(), {broadcasted_bias.value().data_ptr(), bias_multiplier_tensor.value().data_ptr(),
+                                         broadcasted_bias.value().data_ptr(), linear_output.data_ptr()});
+      uids.insert(uids.end(), {'b', 'c', 'd', 'e'});
+      if (kReluFused) {
+        data_ptrs.emplace_back(linear_output.data_ptr()),
+        uids.emplace_back('f');
+      }
+    } else {
+      if (kReluFused) {
+        data_ptrs.emplace_back(linear_output.data_ptr());
+        uids.emplace_back('f');
+      }
+    }
+    auto variantPack = cudnn_frontend::VariantPackBuilder()
+      .setWorkspacePointer(workspace.data_ptr())
+      .setDataPointers(uids.size(), data_ptrs.data())
+      .setUids(uids.size(), uids.data())
+      .build();
+    auto variant_pack_desc = variantPack.get_raw_desc();
+    AT_CUDNN_CHECK(cudnnBackendExecute(handle, plan_desc->get_backend_descriptor(), variant_pack_desc));
+  };
+
+  auto search = execution_plan_cache.find(key);
+  if (search != execution_plan_cache.end()) {
+    cudnn_frontend::ManagedOpaqueDescriptor plan_desc = search->second;
+    run(plan_desc);
+    return;
+  }
+
+  // linear_op computes act_int8 * tranpose(w_int8) (matrix multiplication)
+  // where act_int8 and w_int8 are the input and weight variables, resp.
+  // output is a fp32 tensor
+  auto linear_op = cudnn_frontend::OperationBuilder(CUDNN_BACKEND_OPERATION_MATMUL_DESCRIPTOR)
+      // TODO: make these 2 CUDNN_DATA_INT8 when cudnn enables int8 matmul
+      // .setaMatDesc(cudnn_utils::getTensorDescriptor(input.sizes(), input.strides(), CUDNN_DATA_FLOAT, 'x', key.input_alignment))
+      .setaMatDesc(cudnn_utils::getTensorDescriptor(input_fp.sizes(), input_fp.strides(), CUDNN_DATA_FLOAT, 'x', key.input_alignment))
+      // .setbMatDesc(cudnn_utils::getTensorDescriptor(orig_weight.sizes(), orig_weight.strides(), CUDNN_DATA_FLOAT, 'w', key.weight_alignment))
+      .setbMatDesc(cudnn_utils::getTensorDescriptor(weight_fp.sizes(), weight_fp.strides(), CUDNN_DATA_FLOAT, 'w', key.weight_alignment))
+      .setcMatDesc(cudnn_utils::getTensorDescriptor(linear_output, 'y', key.output_alignment))
+      .setmatmulDesc(getLinearDescriptor(CUDNN_DATA_FLOAT)) // is this right? should it be float?
```

Comment:
I remember we have a table for the descriptor data type, maybe we can implement that in a function: getting descriptor data type from input data type